pkg/envvars/environment.go

```diff
@@ -20,62 +20,119 @@ const (
-	defer func() { _ = gotenv.Apply(strings.NewReader(bashOutput)) }() //nolint:errcheck // we can't do anything with the error
+	LoadConfigFiles(customConfigFiles, workingDirectory)
+}
 
-	env := gotenv.Parse(strings.NewReader(bashOutput))
-	specificShell, ok := env[ShellEnvVarName]
-	if ok {
-		fromSpecificShell := getEnvFromShell(specificShell)
-		_ = gotenv.Apply(strings.NewReader(fromSpecificShell)) //nolint:errcheck // we can't do anything with the error
+// LoadShellEnvironment loads the user's shell environment with special handling of PATHs.
+// First Bash's env is read (as a fallback), then the user's preferred SHELL's env is read.
+// The Bash env PATH is appended to the existing PATH (as a fallback), the user's preferred SHELL's env PATH is prepended (preferential).
+func LoadShellEnvironment() {
+	bashEnvOutput := getEnvFromShell("bash")
+
+	// We first append the Bash PATH as a fallback for if the user's SHELL's env fails to read / parse.
+	// Note: We do this as the more info scraping the better to help ensure we can find binaries for OSS scans.
+	bashEnv := gotenv.Parse(strings.NewReader(bashEnvOutput))
+	if val, ok := bashEnv[PathEnvVarName]; ok {
+		UpdatePath(val, false)
 	}
 
-	// process config files
-	for _, file := range customConfigFiles {
-		if !filepath.IsAbs(file) {
-			file = filepath.Join(workingDirectory, file)
+	// this is applied at the end always, as we do not want it to overwrite existing variables from the user's preferred SHELL
+	defer func() { _ = gotenv.Apply(strings.NewReader(bashEnvOutput)) }() //nolint:errcheck // we can't do anything with the error
+
+	preferredShell, ok := bashEnv[ShellEnvVarName]
+	if ok {
+		preferredShellEnvOutput := getEnvFromShell(preferredShell)
+		_ = gotenv.Apply(strings.NewReader(preferredShellEnvOutput)) //nolint:errcheck // we can't do anything with the error
+
+		preferredShellEnv := gotenv.Parse(strings.NewReader(preferredShellEnvOutput))
+		if val, ok := preferredShellEnv[PathEnvVarName]; ok {
+			UpdatePath(val, true)
 		}
-		loadFile(file)
 	}
 }
 
-func loadFile(fileName string) {
-	// preserve path
-	previousPath := os.Getenv(PathEnvVarName)
+// LoadConfigFiles loads environment variables from configuration files.
+// With special handling for PATH and SDK environment variables.
+// The resultant PATH is constructed as follows:
+// 1. Config file PATH entries (highest precedence)
+// 2. SDK bin directories (if SDK variables like JAVA_HOME, GOROOT are set by the config file)
+// 3. Previous PATH entries (lowest precedence)
+func LoadConfigFiles(customConfigFiles []string, workingDirectory string) {
+	// Capture current SDK environment variables and track their latest values
+	sdkVarNames := []string{"JAVA_HOME", "GOROOT"}
+	sdkValues := make(map[string]string)
+	for _, sdkVar := range sdkVarNames {
+		sdkValues[sdkVar] = os.Getenv(sdkVar)
+		// Unset the env var so we can capture if the config file sets it.
+		_ = os.Unsetenv(sdkVar)
+	}
```

Comment:
There's a few more, but we can cover that later. E.g. `PYTHONPATH`, `GRADLE_HOME`, `MVN_HOME` etc.